pkg/kn/commands/types_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+func TestInitialize(t *testing.T) {
+	p := &KnParams{}
+
+	p.Initialize()
+
+	if reflect.ValueOf(p.ServingFactory).Pointer() != reflect.ValueOf(p.GetConfig).Pointer() {
+		t.Errorf("wrong function assignment detected: %v (expected) != %v (actual)", reflect.ValueOf(p.GetConfig).Pointer(), reflect.ValueOf(p.ServingFactory).Pointer())
+	}
+
+	if reflect.ValueOf(p.NamespaceFactory).Pointer() != reflect.ValueOf(p.CurrentNamespace).Pointer() {
```

Comment:
Same as above. Every test that use reflections to check for internal implementation details has a smell.